ClosedXML/Excel/CalcEngine/XLCalcEngine.cs

```diff
@@ -90,7 +90,7 @@ public IEnumerator GetEnumerator()
         // ** implementation
         private object GetValue(IXLCell cell)
         {
-            if (_evaluating)
+            if (_evaluating || (cell is XLCell && (cell as XLCell).IsEvaluating))
             {
                 throw new InvalidOperationException("Circular Reference");
```

Comment:
Include the cell address (including sheet name) in the exception message. This will allow the user to find the culprit.

Reply:
One of the `cell.Address.ToString(XLReferenceStyle.Default, false)` is what you should include.

Reply:
Sorry, missed this comment